### PR TITLE
DOCS-1415: Fix straightforward yaml errors and defer the complicated ones

### PR DESCRIPTION
**Pull request by girishshankaran**
Product Version(s):

Issue:
https://tigera.atlassian.net/browse/DOCS-1415
https://tigera.atlassian.net/browse/DOCS-1416

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

### CODE REVIEW

File: calico-cloud/network-policy/policy-firewalls/aws-integration/metadata-access.mdx [hunk 1]
```diff
@@ -44,7 +44,7 @@ If the number of pods you need to allow exceeds the number that you need to bloc
     name: tigera-secure
   spec:
     defaultPodMetadataAccess: Allowed
-    <... do not remove any of the existing fields ...>
+    #<... do not remove any of the existing fields ...>
```
Comment (ctauchen) [inline calico-cloud/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: If this is a comment, let's just use regular comment syntax.
```suggestion
    # Do not remove any of the existing fields
```

File: calico-cloud_versioned_docs/version-3.15/network-policy/policy-firewalls/aws-integration/metadata-access.mdx [hunk 2]
```diff
@@ -44,7 +44,7 @@ If the number of pods you need to allow exceeds the number that you need to bloc
     name: tigera-secure
   spec:
     defaultPodMetadataAccess: Allowed
-    <... do not remove any of the existing fields ...>
+    #<... do not remove any of the existing fields ...>
```
Comment (ctauchen) [inline calico-cloud_versioned_docs/version-3.15/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: `#`

File: calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx [hunk 3]
```diff
@@ -24,7 +24,7 @@ can be used to specify the resource type on the CLI:
 
 ## Sample YAML
 
-```yaml
+```yaml noValidation
```
Comment (ctauchen) [inline calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx]: Does this break the validator? It looks fine, and it passes on yamlchecker.com.

I see more than a few of these. Is there something I'm missing?

File: calico-enterprise/reference/resources/policyrecommendations.mdx [hunk 4]
```diff
@@ -29,6 +29,33 @@ $ kubectl patch policyrecommendationscope default -p '{"spec":{"<parameter>":"<v
 
 `$ kubectl patch policyrecommendationscope default -p '{"spec":{"interval":"5m"}}'`
 
+## Sample YAML
```
Comment (ctauchen) [inline calico-enterprise/reference/resources/policyrecommendations.mdx]: Why is this being added?

File: calico-enterprise_versioned_docs/version-3.14/network-policy/policy-firewalls/aws-integration/metadata-access.mdx [hunk 5]
```diff
@@ -44,7 +44,7 @@ If the number of pods you need to allow exceeds the number that you need to bloc
     name: tigera-secure
   spec:
     defaultPodMetadataAccess: Allowed
-    <... do not remove any of the existing fields ...>
+    #<... do not remove any of the existing fields ...>
```
Comment (ctauchen) [inline calico-enterprise_versioned_docs/version-3.14/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: See above

File: calico-enterprise_versioned_docs/version-3.15/operations/troubleshoot/commands.mdx [hunk 6]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: default
```
Comment (ctauchen) [inline calico-enterprise_versioned_docs/version-3.15/operations/troubleshoot/commands.mdx]: This seems to be a change. Do we need to get SME review for this?

File: calico-enterprise_versioned_docs/version-3.16/network-policy/policy-firewalls/aws-integration/metadata-access.mdx [hunk 7]
```diff
@@ -44,7 +44,7 @@ If the number of pods you need to allow exceeds the number that you need to bloc
     name: tigera-secure
   spec:
     defaultPodMetadataAccess: Allowed
-    <... do not remove any of the existing fields ...>
+    #<... do not remove any of the existing fields ...>
```
Comment (ctauchen) [inline calico-enterprise_versioned_docs/version-3.16/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: see above

File: calico/operations/troubleshoot/commands.mdx [hunk 8]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: default
```
Comment (ctauchen) [inline calico/operations/troubleshoot/commands.mdx]: Change?

File: calico_versioned_docs/version-3.24/operations/troubleshoot/commands.mdx [hunk 9]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: my-installation
```
Comment (ctauchen) [inline calico_versioned_docs/version-3.24/operations/troubleshoot/commands.mdx]: What's this change about?

File: calico_versioned_docs/version-3.25/operations/troubleshoot/commands.mdx [hunk 10]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: default
```
Comment (ctauchen) [inline calico_versioned_docs/version-3.25/operations/troubleshoot/commands.mdx]: Change?

**Review by ctauchen — COMMENTED** [review]
What a job! Thanks, @girishshankaran. This will get us up and running with the YAML validator.

There a few instances of code comments of the type <... bla bla ..>. I think rather than commenting them out, we should convert them to comments. 

In a few places I noticed new or changed content. Please explain if they're intentional or fix if they're not.

File: calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx (hunk 3 above)
Comment (girishshankaran) [reply calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx]: Yes Chris. I added the `noValidation` tag to some of the yaml blocks that were actually correct according to the [tool](https://codebeautify.org/yaml-validator/) I was using. However, I added the tag because the validator was complaining.

File: calico-enterprise/reference/resources/policyrecommendations.mdx (hunk 4 above)
Comment (girishshankaran) [reply calico-enterprise/reference/resources/policyrecommendations.mdx]: I need to investigate this. This is the file where I had a merge conflict.

File: calico-cloud/network-policy/policy-firewalls/aws-integration/metadata-access.mdx (hunk 1 above)
Comment (girishshankaran) [reply calico-cloud/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: Fixed

File: calico-cloud_versioned_docs/version-3.15/network-policy/policy-firewalls/aws-integration/metadata-access.mdx (hunk 2 above)
Comment (girishshankaran) [reply calico-cloud_versioned_docs/version-3.15/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: Fixed.

File: calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx (hunk 3 above)
Comment (girishshankaran) [reply calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx]: Removed the tag for testing.

File: calico-enterprise_versioned_docs/version-3.15/operations/troubleshoot/commands.mdx (hunk 6 above)
Comment (girishshankaran) [reply calico-enterprise_versioned_docs/version-3.15/operations/troubleshoot/commands.mdx]: Reverted the change. Fixed the yaml with an indentation change.

File: calico/operations/troubleshoot/commands.mdx (hunk 8 above)
Comment (girishshankaran) [reply calico/operations/troubleshoot/commands.mdx]: Reverted the change. Fixed the yaml with an indentation change.

File: calico_versioned_docs/version-3.24/operations/troubleshoot/commands.mdx (hunk 9 above)
Comment (girishshankaran) [reply calico_versioned_docs/version-3.24/operations/troubleshoot/commands.mdx]: Reverted the change. Fixed the yaml with an indentation change.

File: calico_versioned_docs/version-3.25/operations/troubleshoot/commands.mdx (hunk 10 above)
Comment (girishshankaran) [reply calico_versioned_docs/version-3.25/operations/troubleshoot/commands.mdx]: Reverted the change. Fixed the yaml with an indentation change.

File: calico-enterprise_versioned_docs/version-3.14/network-policy/policy-firewalls/aws-integration/metadata-access.mdx (hunk 5 above)
Comment (girishshankaran) [reply calico-enterprise_versioned_docs/version-3.14/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: Fixed

File: calico-enterprise_versioned_docs/version-3.16/network-policy/policy-firewalls/aws-integration/metadata-access.mdx (hunk 7 above)
Comment (girishshankaran) [reply calico-enterprise_versioned_docs/version-3.16/network-policy/policy-firewalls/aws-integration/metadata-access.mdx]: Fixed

File: calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx (hunk 3 above)
Comment (girishshankaran) [reply calico-cloud_versioned_docs/version-3.15/reference/resources/globalalert.mdx]: Test passed without the tag :).

File: calico-enterprise/reference/resources/policyrecommendations.mdx (hunk 4 above)
Comment (girishshankaran) [reply calico-enterprise/reference/resources/policyrecommendations.mdx]: Fixed this.

File: calico-enterprise/operations/troubleshoot/commands.mdx [hunk 11]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: default
```
Comment (ctauchen) [inline calico-enterprise/operations/troubleshoot/commands.mdx]: Why this change?

File: calico-enterprise_versioned_docs/version-3.14/operations/troubleshoot/commands.mdx [hunk 12]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: tigera-operator
```
Comment (ctauchen) [inline calico-enterprise_versioned_docs/version-3.14/operations/troubleshoot/commands.mdx]: Why this change?

File: calico-enterprise_versioned_docs/version-3.16/operations/troubleshoot/commands.mdx [hunk 13]
```diff
@@ -237,8 +237,8 @@ items:
 - apiVersion: operator.tigera.io/v1
   kind: Installation
   metadata:
-    - apiVersion: operator.tigera.io/v1
- spec:
+    name: default
```
Comment (ctauchen) [inline calico-enterprise_versioned_docs/version-3.16/operations/troubleshoot/commands.mdx]: Why this change?

**Review by ctauchen — COMMENTED** [review]
A few more for you @girishshankaran 

Might be a good idea to double-check that each of these and the previous changes are ok across all product/versions, just to be sure.

File: calico-enterprise/operations/troubleshoot/commands.mdx (hunk 11 above)
Comment (girishshankaran) [reply calico-enterprise/operations/troubleshoot/commands.mdx]: Reverted the change. Fixed by adjusting the indentation.

Comment (girishshankaran) [reply calico-enterprise/operations/troubleshoot/commands.mdx]: I seem to have fixed this yesterday. Somehow the change got lost.

File: calico-enterprise_versioned_docs/version-3.14/operations/troubleshoot/commands.mdx (hunk 12 above)
Comment (girishshankaran) [reply calico-enterprise_versioned_docs/version-3.14/operations/troubleshoot/commands.mdx]: Fixed it again.

File: calico-enterprise_versioned_docs/version-3.16/operations/troubleshoot/commands.mdx (hunk 13 above)
Comment (girishshankaran) [reply calico-enterprise_versioned_docs/version-3.16/operations/troubleshoot/commands.mdx]: Fixed it.

**Comment by ctauchen** [comment]
LGTM
